static/app/views/monitors/components/monitorForm.tsx

```diff
@@ -154,7 +159,10 @@ function mapFormErrors(responseJson?: any) {
   // Bring nested config entries to the top
   const {config, ...responseRest} = responseJson;
   const configErrors = Object.fromEntries(
-    Object.entries(config).map(([key, value]) => [`config.${key}`, value])
+    Object.entries(config).map(([key, value]) => [
+      `config.${camelToUnderscore(key)}`,
+      value,
+    ])
```

Comment:
I think we can just use lodash [`snakeCase`](https://lodash.com/docs/4.17.15#snakeCase) here for now.

```suggestion
    Object.entries(config).map(([key, value]) => [`config.${snakeCase(key)}`, value])
```